.github/workflows/sbt-tests.yml

```diff
@@ -31,3 +31,42 @@ jobs:
       run: sbt licenseCheck
     - name: Run tests with coverage
       run: sbt jacoco
+
+  generated-files-validation:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up JDK 11
+      uses: actions/setup-java@v4
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: 'sbt'
+
+    - name: Generate OpenAPI Client and Server Model classes
+      run: sbt generate
+
+    - name: Check Generated Files Differences
+      run: |
+        CLIENT_DIFF=$(git diff --name-only clients/java)
+        SERVER_DIFF=$(git diff --name-only server/src/main/java/io/unitycatalog/server/model)
+        
+        if [[ -n $CLIENT_DIFF || -n $SERVER_DIFF ]]; then
+          echo "This PR contains changes to generated files. Making changes to generated files is not allowed."
```

Comment:
this 
"This PR contains changes to generated files that will be overridden when `build/sbt generate` is run again. If you have changed the UC REST APIs such that changes to generated files are expected, then please make sure all generated files have been properly updated in this PR. Otherwise, without changes to UC REST APIs, manual changes to the generated files are not acceptable."